lib/ansible/module_utils/aws/acm.py

```diff
@@ -113,13 +120,15 @@ def get_certificates(self, client, module, domain_name=None, statuses=None, arn=
             except (BotoCoreError, ClientError) as e:
                 module.fail_json_aws(e, msg="Couldn't obtain certificate metadata for domain %s" % certificate['DomainName'])
 
-            # in some states, ACM resources do not have a corresponding cert
-            if cert_data['Status'] not in ['PENDING_VALIDATION', 'VALIDATION_TIMED_OUT', 'FAILED']:
-                try:
-                    cert_data.update(self.get_certificate_with_backoff(client, certificate['CertificateArn']))
-                except (BotoCoreError, ClientError, KeyError) as e:
-                    if e.response['Error']['Code'] != "RequestInProgressException":
-                        module.fail_json_aws(e, msg="Couldn't obtain certificate data for domain %s" % certificate['DomainName'])
+            try:
+                cert = self.get_certificate_with_backoff(client, certificate['CertificateArn'])
+            except (BotoCoreError, ClientError) as e:
+                if e.response['Error']['Code'] != "RequestInProgressException":
```

Comment:
If we get to this point and the error code is a RequestInProgressException, cert will be undefined so `if cert:` below will fail. You may want a try -> except -> else pattern (`else` executes if no exception was raised).

Also, BotocoreError exceptions don't have a 'response' attribute (ClientError does) so this line could cause an AttributeError. You can use the [is_boto3_error_code](https://docs.ansible.com/ansible/2.9/dev_guide/platforms/aws_guidelines.html#using-is-boto3-error-code), e.g.:
```
from ansible.module_utils.aws.core import is_boto3_error_code
...
try:
    cert = self.get_certificate_with_backoff(client, certificate['CertificateArn'])
except is_boto3_error_code('RequestInProgressException'):
    pass
except (BotoCoreError, ClientError) as e:  # pylint: disable=duplicate-except
    # Hard fail for all exceptions which do not have the error code 'RequestInProgressException'
    module.fail_json_aws(e, msg="Couldn't obtain certificate data for domain %s" % certificate['DomainName'])
else:
    cert_data.update(cert)
```

 or separate the BotoCoreError and ClientError exceptions as the second example here shows https://docs.ansible.com/ansible/2.9/dev_guide/platforms/aws_guidelines.html#using-fail-json-aws.

Reply:
Hmm, I'm not sure what the `RequestInProgressException` bit is actually for. I think I just copied and pasted that chunk from the origin `aws_acm_info` module. I guess I can leave the functionality as is (and fix the undefined error.)

Reply:
Ok, I've committed a fix for this.